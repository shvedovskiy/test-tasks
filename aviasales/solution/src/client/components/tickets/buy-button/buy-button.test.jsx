/* eslint-disable no-undef */
import React from 'react';

import BuyButton from './buy-button';


describe('<BuyButton />', () => {
  const props = {
    buyClick: jest.fn(),
  };
  const buyButton = shallow(<BuyButton {...props}>12500$</BuyButton>);

  it('renders properly', () => {
    expect(buyButton).toMatchSnapshot();
  });
});
