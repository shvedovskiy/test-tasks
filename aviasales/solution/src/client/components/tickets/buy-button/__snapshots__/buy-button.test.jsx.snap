// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BuyButton /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BuyButton
    buyClick={[MockFunction]}
  >
    12500$
  </BuyButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled__BuyButtonText>
          Купить
        </styled__BuyButtonText>,
        <styled__BuyButtonText>
          за 
          12 500$
        </styled__BuyButtonText>,
      ],
      "onClick": [MockFunction],
      "tabIndex": "0",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Купить",
        },
        "ref": null,
        "rendered": "Купить",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "за ",
            "12 500$",
          ],
        },
        "ref": null,
        "rendered": Array [
          "за ",
          "12 500$",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled__BuyButtonText>
            Купить
          </styled__BuyButtonText>,
          <styled__BuyButtonText>
            за 
            12 500$
          </styled__BuyButtonText>,
        ],
        "onClick": [MockFunction],
        "tabIndex": "0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Купить",
          },
          "ref": null,
          "rendered": "Купить",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "за ",
              "12 500$",
            ],
          },
          "ref": null,
          "rendered": Array [
            "за ",
            "12 500$",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
